spec/unit/ui_spec.rb

```diff
@@ -91,4 +96,23 @@
       Machinery::Ui.warn("foo")
     end
   end
+
+  describe ".reset_line" do
+    it "sends reset escape sequences if there progress output that needs to be cleared" do
+      allow($stdout).to receive(:tty?).and_return(true)
+      allow(Machinery::Ui).to receive(:use_pager).and_return(false)
+      expect(STDOUT).to receive(:print).exactly(4).times # 1 for progress, 1 for print, 2 for reset
```

Comment:
Wouldn't it be better to just explicitly check for the four calls and their exact arguments?